spark/core/src/main/scala/org/elasticsearch/spark/cfg/SparkSettings.java

```diff
@@ -51,6 +53,13 @@ public SparkSettings(SparkConf cfg) {
         String appId = cfg.get("spark.app.id", "");
         String opaqueId = String.format(Locale.ROOT, "[spark] [%s] [%s] [%s]", user, appName, appId);
         this.setOpaqueId(opaqueId);
+        this.setUserAgent(String.format(
+                        Locale.ROOT,
+                        "elasticsearch-hadoop/%s spark/%s",
+                        Version.versionNumber(),
+                        SparkContext.getOrCreate(cfg).version()
```

Comment:
Is there a lighter way to get this version or are we sure that the SparkContext will already be present when getting it? Would creating the context here and not closing it out be problematic?

Reply:
Yeah it does. I haven't figured out how to access that from Java yet though (it's not as straightforward as I'd have thought). I'm currently changing it to use a helper scala class to expose it to java.